warehouse/query-core/src/test/java/datawave/query/util/SummaryTest.java

```diff
@@ -0,0 +1,321 @@
+                        Set.of("CONTENT_SUMMARY:CONTENT: You can get much farther with a kind word and a gu: : [] 9223372036854775807 false"));
+
+        runTestQuery(queryString, format.parse("19000101"), format.parse("20240101"), extraParameters, goodResults);
+    }
+
+    @Test
+    public void testSize() throws Exception {
+        Map<String,String> extraParameters = new HashMap<>();
+        extraParameters.put("include.grouping.context", "true");
+        extraParameters.put("return.fields", "CONTENT_SUMMARY");
+        extraParameters.put("query.syntax", "LUCENE");
+
+        String queryString = "QUOTE:(farther) #SUMMARY(SIZE:50)";
+
+        // not sure why the timestamp and delete flag are present
+        Set<String> goodResults = new HashSet<>(
+                        Set.of("CONTENT_SUMMARY:CONTENT: You can get much farther with a kind word and a gu: : [] 9223372036854775807 false"));
+
+        runTestQuery(queryString, format.parse("19000101"), format.parse("20240101"), extraParameters, goodResults);
+    }
+
+    @Test
+    public void testOverMaxSize() throws Exception {
+        Map<String,String> extraParameters = new HashMap<>();
+        extraParameters.put("include.grouping.context", "true");
+        extraParameters.put("return.fields", "CONTENT_SUMMARY");
+        extraParameters.put("query.syntax", "LUCENE");
+
+        String queryString = "QUOTE:(farther) #SUMMARY(SIZE:90000)";
+
+        // not sure why the timestamp and delete flag are present
+        Set<String> goodResults = new HashSet<>(Set.of(
+                        "CONTENT_SUMMARY:CONTENT: You can get much farther with a kind word and a gun than you can with a kind word alone: : [] 9223372036854775807 false"));
+
+        runTestQuery(queryString, format.parse("19000101"), format.parse("20240101"), extraParameters, goodResults);
+    }
+
+    @Test
+    public void testNegativeSize() throws Exception {
+        Map<String,String> extraParameters = new HashMap<>();
+        extraParameters.put("include.grouping.context", "true");
+        extraParameters.put("return.fields", "CONTENT_SUMMARY");
+        extraParameters.put("query.syntax", "LUCENE");
+
+        String queryString = "QUOTE:(farther) #SUMMARY(SIZE:-50)";
+
+        // not sure why the timestamp and delete flag are present
+        Set<String> goodResults = new HashSet<>(Set.of("CONTENT_SUMMARY:CONTENT: Y: : [] 9223372036854775807 false"));
+
+        runTestQuery(queryString, format.parse("19000101"), format.parse("20240101"), extraParameters, goodResults);
+    }
+
+    @Test
+    public void testNoContentFound() throws Exception {
+        Map<String,String> extraParameters = new HashMap<>();
+        extraParameters.put("include.grouping.context", "true");
+        extraParameters.put("return.fields", "CONTENT_SUMMARY");
+        extraParameters.put("query.syntax", "LUCENE");
+
+        String queryString = "QUOTE:(farther) #SUMMARY(SIZE:50/ONLY/VIEWS:CANTFINDME,ORME)";
```

Comment:
I would like to see some more tests surrounding the configuration so there are concrete examples of how the iterator behaves with different inputs. 

It seems odd that if I specify a view name that is additive to some default view names. 

You should not have to specify 'CONTENT_SUMMARY' in the return fields in order to get a summary back. So really you need to handle two distinct cases
1) user does not specify 'return.fields', nothing to do
2) user specifies 'return.fields', need to add 'CONTENT_SUMMARY'

Reply:
1.  will try and add more tests
2. incase the default ones do not cover everything and it also allows you to change the priority
3. you dont. it seems that is just for the test so we only get back the fields we care about (see ExcerptTest).

Reply:
done